tests/test_post_gen_project.py

```
@@ -0,0 +1,184 @@
+
+
+@pytest.mark.parametrize(
+    "templates, expected_template",
+    [
+        # test if copy_journal_template_files can find the expected template.
+        # C1: with multiple templates, expect no NotADiectoryError.
+        (["article", "foo", "bar"], "article")
+    ],
+)
+def test_copy_journal_template_files_find_template(
+    user_filesystem, templates, expected_template
+):
+    home_path = user_filesystem
+    skm_path = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    cwd_path = home_path / "cwd_dir"
+    for template in templates:
+        template_path = skm_path / "templates" / template
+        template_path.mkdir(parents=True, exist_ok=True)
+    copy_journal_template_files(expected_template, cwd_path)
+
+
+@pytest.mark.parametrize(
+    "templates, expected_template",
+    [
+        # test copy_journal_template_files when template is not existed.
+        # C1: empty templates, expect NotADirectoryError.
+        ([], "article"),
+        # C2: template does not exist, expect NotADirectoryError.
+        (["foo", "bar"], "article"),
+    ],
+)
+def test_copy_journal_template_files_find_template_bad(
+    user_filesystem, templates, expected_template
+):
+    home_path = user_filesystem
+    skm_path = home_path / ".cookiecutters" / "scikit-package-manuscript"
+    skm_path.mkdir(parents=True, exist_ok=True)
+    cwd_path = home_path / "cwd_dir"
+    for template in templates:
+        template_path = skm_path / "templates" / template
+        template_path.mkdir(parents=True, exist_ok=True)
+
+    with pytest.raises(
+        NotADirectoryError,
+        match=r"Cannot find the provided journal_tamplate: ",
+    ):
+        cwd_path = copy_journal_template_files(expected_template, cwd_path)
+
+
+@pytest.mark.parametrize(
+    "template_files",
+    [
+        # test copy_journal_template_files with different journal
+        #   template structure.
+        # C1: empty template, expect nothing happens.
```

I would rather we raise an exception for this case.